src/main/java/com/github/kokorin/jaffree/process/ProcessHandler.java

```diff
@@ -180,7 +199,9 @@ protected T interactWithProcess(final Process process) {
             status = process.waitFor();
             LOGGER.info("Process has finished with status: {}", status);
 
-            waitForExecutorToStop(executor, EXECUTOR_TIMEOUT_MILLIS);
+            if (executorTimeoutMillis > 0) {
```

Comment:
`waitForExecutorToStop` must be invoked, otherwise ProcessHandler exists immediately.
Check for zero timeout should be done in that method.

Reply:
Let's discuss the implementation of `waitForExecutorToStop()`.

First of all I think that this line
```
            if (System.currentTimeMillis() - waitStarted > timeoutMillis) {
```
has an edge-case bug.
It should actually use `>=`.

Think about the cause where `timeoutMillis` == 0:

- `System.currentTimeMillis() - waitStarted` will almost certainly be 0
- The if clause will fail and you will end up invoking `Thread.sleep(100)`
- That would be unintended / unexpected

In a practical sense, this was not important when we had a constant EXECUTOR_TIMEOUT_MILLIS, and it is not important if we short-circuit for the `timeoutMillis==0` case.
But given that we are discussing calling `waitForExecutorToStop()` for the zero case, I thought I should mention it.

Reply:
Second of all I wonder why you have a `do ... while` loop instead of a `while` loop.

The `do ... while` deals with the case where the executor has already stopped by the time we run `waitForExecutorToStop()`.

In this case, why do we want to force a `Thread.sleep(100)`?
Why would we want to log "Executor hasn't yet stopped..." when that is not true?

The log is only at TRACE level, so the practical outcome is pretty harmless.
Further it is perhaps very unlikely to hit this case ... but it seems to me that a `while` loop would be the correct implementation - am I misunderstanding something?

Reply:
> waitForExecutorToStop must be invoked, otherwise ProcessHandler exists immediately.

This was confusing me for a long time ... until I realised ... I think I misunderstood your previous request:

> I think it makes sense to allow a user to disable timeout by setting 0 (zero) value.

What you meant was "a zero value means wait indefinitely". 
I thought you meant "a zero value means don't wait at all".

As a general rule in configuration values - I prefer to let zero mean zero - and then use -1, or "any negative value" to mean a special case like "disable the timeout". 
The bonus with "any negative value" is that you don't need to check for valid values - all values have some meaning.

Reply:
> However, in this particular case, would an actual `0ms` timeout ever be a useful setting? Perhaps here the "0" as a special value makes sense ...

Even ffmpeg itself uses `-threads 0` as an option to use all threads available.

> Second of all I wonder why you have a do ... while loop instead of a while loop.
> The do ... while deals with the case where the executor has already stopped by the time we run waitForExecutorToStop().

To be honest, I don't care. It's highly unlikely that ffmpeg will complete in 100ms or less.

> `if (System.currentTimeMillis() - waitStarted > timeoutMillis) {`

I think it can be like  `if (timeoutMillis> 0 && System.currentTimeMillis() - waitStarted > timeoutMillis) {`